Gems/Atom/Tools/ShaderManagementConsole/Scripts/GenerateShaderVariantListForAllShaders.py

```diff
@@ -0,0 +1,44 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import re
+import json
+import GenerateShaderVariantListUtil
+
+def search_and_save_shadervariant(searchPath):
+    for subdir, dirs, files in os.walk(searchPath):
+        for file in files:
+            filename = os.path.join(subdir, file)
+            if re.search('^.*\.shader$', filename):
+                shaderVariantList, defaultShaderVariantListFilePath = GenerateShaderVariantListUtil.create_shadervariantlist_for_shader(filename)
```

Comment:
This process probably isn't the most efficient. This is looping through all the shaders, then searching for all materials that use that shader, load the material, and generate the variant list for that one shader. So if a material type has 10 shaders, then every material of that type will be loaded 10 times. 

Instead, it would probably be better to scan for all materials, load each material once, and then collect information about the shaders and variants in memory. Then in the end, dump out all the shader variant lists.